src/PowerShellEditorServices/Services/Analysis/AnalysisService.cs

```diff
@@ -349,369 +164,36 @@ public IEnumerable<string> GetPSScriptAnalyzerRules()
-                } else {
-                    sessionState = InitialSessionState.CreateDefault2();
-                }
-                sessionState.ImportPSModule(new [] { pssaModuleInfo.ModuleBase });
-
-                // RunspacePool takes care of queuing commands for us so we do not
-                // need to worry about executing concurrent commands
-                return RunspaceFactory.CreateRunspacePool(sessionState);
-            }
-        }
-
-        #endregion //private methods
-
-        #region IDisposable Support
-
-        private bool _disposedValue = false; // To detect redundant calls
-
-        /// <summary>
-        /// Dispose of this object.
-        /// </summary>
-        /// <param name="disposing">True if the method is called by the Dispose method, false if called by the finalizer.</param>
-        protected virtual void Dispose(bool disposing)
-        {
-            if (!_disposedValue)
-            {
-                if (disposing)
-                {
-                    _analysisRunspacePool.Dispose();
-                    _analysisRunspacePool = null;
-                }
-
-                _disposedValue = true;
-            }
-        }
-
-        /// <summary>
-        /// Clean up all internal resources and dispose of the analysis service.
-        /// </summary>
-        public void Dispose()
-        {
-            // Do not change this code. Put cleanup code in Dispose(bool disposing) above.
-            Dispose(true);
-        }
-
-        #endregion
-
-        /// <summary>
-        /// Wraps the result of an execution of PowerShell to send back through
-        /// asynchronous calls.
-        /// </summary>
-        private class PowerShellResult
+        public async Task RunScriptDiagnosticsAsync(
+            ScriptFile[] filesToAnalyze,
+            CancellationToken token)
         {
-            public PowerShellResult(
-                PSObject[] output,
-                ErrorRecord[] errors,
-                bool hasErrors)
-            {
-                Output = output;
-                Errors = errors;
-                HasErrors = hasErrors;
-            }
+            // This token will be cancelled (typically via LSP cancellation) if the token passed in is cancelled or if
+            // multiple requests come in (the last one wins).
+            CancellationToken ct;
 
-            public PSObject[] Output { get; }
-
-            public ErrorRecord[] Errors { get; }
-
-            public bool HasErrors { get; }
-        }
-
-        internal async Task RunScriptDiagnosticsAsync(
-            ScriptFile[] filesToAnalyze)
-        {
             // If there's an existing task, attempt to cancel it
             try
             {
-                if (s_existingRequestCancellation != null)
-                {
-                    // Try to cancel the request
-                    s_existingRequestCancellation.Cancel();
+                await _existingRequestCancellationLock.WaitAsync();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/app/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4416818)